src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -0,0 +1,128 @@
+import logging
+import sentry_sdk
+
+from rest_framework.response import Response
+from rest_framework.exceptions import ParseError
+
+from sentry import features, eventstore
+from sentry.api.bases import OrganizationEventsEndpointBase, NoProjects
+from sentry.snuba import discover
+
+
+logger = logging.getLogger(__name__)
+
+
+def find_event(items, function, default=None):
+    return next((item for item in items if function(item)), default)
```

Comment:
Maybe name it `predicate`? 

```suggestion
def find_event(items, predicate, default=None):
    return next((item for item in items if predicate(item)), default)
```

Reply:
Mm was trying to match the filter docs, but.....
🤔 no clue why I didn't do this...  i'm gonna change this to: 
```python
return next(filter(function, items), default)
```